docs/Documentation/Software_Tools/spack.md

```diff
@@ -0,0 +1,306 @@
+
+
+
+## Installed packages
+
+`spack find	                 ` list all locally installed packages 
+
+`spack find --deps <package> ` list dependencies of <package> 
+
+`spack find --explicit	     ` list packages that were explicitly requested via spack install 
+
+`spack find --implicit	     ` list packages that were installed as a dependency to an explicitly installed package
+
+`spack find --long	         ` include partial hash in package listing. Useful to see distinct builds 
+
+`spack find --paths	         ` show installation paths                         
+
+
+Finding how an installed package was built does not seem as straightforward as it should be. 
+Probably the best way is to examine `<install_path>/.spack/build.env`, where `<install_path>` is the Spack-created directory with the hash for the package being queried. 
+The environment variable `SPACK_SHORT_SPEC` in `build.env` contains the Spack command that can be used to recreate the package (including any implicitly defined variables, e.g., arch). 
+The 7-character short hash is also included, and should be excluded from any spack install command.
+
+
+`@	         ` package versions. Can use range operator “:”, e.g., X@1.2:1.4 . Range is inclusive and open-ended, e.g., “X@1.4:” matches any version of package X 1.4 or higher. 
+
+`%	         ` compiler spec. Can include versioning, e.g., X%gcc@4.8.5 
+
+`+,-,~       ` build options. +opt, -opt, “~” is equivalent to “-“ 
+
+`name=value	 ` build options for non-Boolean flags. Special names are cflags, cxxflags, fflags, cppflags, ldflags, and ldlibs 
+
+`target=valu ` for defined CPU architectures, e.g., target=haswell          
+
+`os=value	 ` for defined operating systems                                    
+
+`^	         ` dependency specification, using above specs as appropriate       
+
+`^/<hash>	 ` specify dependency where <hash> is of sufficient length to resolve uniquely 
+
+## External Packages
+
+Sometimes dependencies are expected to be resolved through a package that is installed as part of the host system, or otherwise outside of the Spack database. 
+One example is Slurm integration into MPI builds. 
+If you were to try to add a dependency on one of the listed Slurms in the Spack database, you might see, e.g.,
+
+```
+[$user@el2 ~]$ spack spec openmpi@3.1.3%gcc@7.3.0 ^slurm@19-05-3-2
+Input spec
+--------------------------------
+openmpi@3.1.3%gcc@7.3.0
+    ^slurm@19-05-3-2
+
+Concretized
+--------------------------------
+==> Error: The spec 'slurm' is configured as not buildable, and no matching external installs were found
+```
+
+Given that something like Slurm is integrated deeply into the runtime infrastructure of our local environment, we really want to point to the local installation. 
+The way to do that is with a `packages.yaml` file, which can reside in the standard Spack locations (see Configuration below). 
+See the Spack docs on external packages for more detail. 
```

Comment:
I would add a hyperlink here for spack docs. I don't think one was added earlier.

Reply:
fixed